src/memory_region.rs

```diff
@@ -1281,38 +1001,19 @@ mod test {
         };
         let mem1 = [0x11, 0x22];
         let mem2 = [0x33];
-        let mem3 = [0x44, 0x55, 0x66];
-        let mem4 = [0x77, 0x88, 0x99];
         let m = MemoryMapping::new(
             vec![
                 MemoryRegion::new_readonly(&mem1, ebpf::MM_INPUT_START),
                 MemoryRegion::new_readonly(&mem2, ebpf::MM_INPUT_START + mem1.len() as u64),
-                MemoryRegion::new_readonly(
-                    &mem3,
-                    ebpf::MM_INPUT_START + (mem1.len() + mem2.len()) as u64,
-                ),
-                MemoryRegion::new_readonly(
-                    &mem4,
-                    ebpf::MM_INPUT_START + (mem1.len() + mem2.len() + mem3.len()) as u64,
-                ),
             ],
             &config,
             SBPFVersion::V3,
         )
         .unwrap();
 
         assert_eq!(m.load::<u16>(ebpf::MM_INPUT_START).unwrap(), 0x2211);
-        assert_eq!(m.load::<u32>(ebpf::MM_INPUT_START).unwrap(), 0x44332211);
-        assert_eq!(
-            m.load::<u64>(ebpf::MM_INPUT_START).unwrap(),
-            0x8877665544332211
-        );
-        assert_eq!(m.load::<u16>(ebpf::MM_INPUT_START + 1).unwrap(), 0x3322);
-        assert_eq!(m.load::<u32>(ebpf::MM_INPUT_START + 1).unwrap(), 0x55443322);
-        assert_eq!(
-            m.load::<u64>(ebpf::MM_INPUT_START + 1).unwrap(),
-            0x9988776655443322
-        );
```

Comment:
Are we not testing these happy paths anymore?

Reply:
They were testing cross region accesses, which this PR stops supporting.

Reply:
Is this going to impact existing programs?

Reply:
Currently we are only using the aligned memory mapping and that does not support cross region accesses either.